daiquiri/oai/adapter.py

```diff
@@ -48,3 +68,149 @@ def get_sample_identifier(self):
+            and (table.schema.published is not None)
+
+        return table.pk, identifier, datestamp, set_spec, public
+
+
+class DoiMetadataOaiAdapterMixin(MetadataOaiAdapterMixin):
+
+    def get_schema_record(self, schema):
+        identifier = self.get_identifier(schema.doi)
+        datestamp = schema.updated or schema.published
+        set_spec = 'schema'
+        public = (schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(schema.published) \
+            and bool(schema.doi)
+
+        return schema.pk, identifier, datestamp, set_spec, public
+
+    def get_table_record(self, table):
+        identifier = self.get_identifier(table.doi)
+        datestamp = table.updated or table.published
+        set_spec = 'table'
+        public = (table.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(table.published) \
+            and bool(table.doi) \
+            and (table.schema.metadata_access_level == ACCESS_LEVEL_PUBLIC) \
+            and bool(table.schema.published)
+
+        return table.pk, identifier, datestamp, set_spec, public
+
+
+class RegistryOaiAdapterMixin(object):
+
+    services = ['registry', 'authority', 'web', 'tap', 'conesearch']
+
+    def get_oai_dc_service_serializer_class(self):
+        return import_class('daiquiri.registry.serializers.DublincoreSerializer')
+
+    def get_ivo_vor_service_serializer_class(self):
+        return import_class('daiquiri.registry.serializers.VoresourceSerializer')
+
+    def get_service_list(self):
+        for i, service in enumerate(self.services):
+            yield 'service', self.get_service(i + 1)
+
+    def get_service(self, pk):
+        index = pk - 1
+
+        if self.services[index] == 'registry':
+            from daiquiri.registry.vo import get_resource
+            return get_resource()
+        elif self.services[index] == 'authority':
+            from daiquiri.registry.vo import get_authority_resource
+            return get_authority_resource()
+        elif self.services[index] == 'web':
+            from daiquiri.registry.vo import get_web_resource
+            return get_web_resource()
+        elif self.services[index] == 'tap' and apps.is_installed('daiquiri.tap'):
+            from daiquiri.tap.vo import get_resource
+            return get_resource()
+        elif self.services[index] == 'conesearch' and apps.is_installed('daiquiri.conesearch'):
+            from daiquiri.conesearch.vo import get_resource
+            return get_resource()
+
+    def get_service_record(self, service):
+        index = self.services.index(service['service']) + 1
+        identifier = service['identifier']
+        datestamp = settings.SITE_UPDATED
+        set_spec = 'ivo_managed'
+        public = True
+
+        return index, identifier, datestamp, set_spec, public
+
+
+class DefaultOaiAdapter(RegistryOaiAdapterMixin, DoiMetadataOaiAdapterMixin, BaseOaiAdapter):
+
```

Comment:
This is also misleading that for the OaiAdapter the resource_types are a dictionary and not a list like for the other Adapters. We could somehow think how to address this inconsistency.

Reply:
Fixed it (and introduced a get_metadata_prefixes "thing")